### PR TITLE
Add fuzzy query

### PR DESCRIPTION
**Pull request by fatal69100**
To reproduce the previous behavior : -> Request elasticsearch in `PREFIX` mode first (fast and confident)
-> If no results found: request in `FUZZY` mode (with ngrams)

### CODE REVIEW

File: libs/mimir/src/adapters/primary/bragi/handlers.rs [hunk 1]
```diff
@@ -44,30 +48,89 @@ pub struct InternalError {
 impl Reject for InternalError {}
 
 #[instrument(skip(client, settings))]
-pub async fn forward_geocoder<S>(
+pub async fn forward_geocoder(
     params: ForwardGeocoderQuery,
     geometry: Option<Geometry>,
-    client: S,
+    client: ElasticsearchStorage,
```
Comment (crocme10) [inline libs/mimir/src/adapters/primary/bragi/handlers.rs:54]: Why do you use ElasticsearchStorage instead of a generic?

Comment (fatal69100) [reply libs/mimir/src/adapters/primary/bragi/handlers.rs:54]: It was easier for me to understand and to develop this feature with explicit code. I am not sure, but do we use another client than ElasticsearchStorage ?

File: libs/mimir/src/adapters/primary/bragi/handlers.rs [hunk 2]
```diff
@@ -44,30 +48,89 @@ pub struct InternalError {
...
+                }
+            } else {
+                Ok(with_status(json(&resp), StatusCode::OK))
+            }
+        }
+        Err(err) => Err(warp::reject::custom(InternalError {
+            reason: InternalErrorReason::ObjectNotFoundError,
+            info: err.to_string(),
+        })),
+    }
```
Comment (pbench) [inline libs/mimir/src/adapters/primary/bragi/handlers.rs:121]: You don't run a fuzzy query when :

- the prefix query failed
- the prefix query succeeded but found nothing.

Is that on purpose ? I would expect a fuzzy query to be launched when one of these things happen.

**Comment by pbench** [comment]
Related to https://github.com/CanalTP/mimirsbrunn/pull/688 
I guess we will need to have different timeouts for prefix/fuzzy queries.
This can be done in a subsequent PR

File: libs/mimir/src/adapters/primary/bragi/handlers.rs (hunk 2 above)
Comment (pbench) [reply libs/mimir/src/adapters/primary/bragi/handlers.rs:121]: Ok, this is the same behavior as with es 2 https://github.com/CanalTP/mimirsbrunn/blob/da72c2fac5887d3c6fbbc2b4d69201ecb9a95319/libs/bragi/src/query.rs#L690

**Pull request merged**